frontend/packages/kubevirt-plugin/src/components/vms/vm-console.tsx

```diff
@@ -76,11 +79,11 @@ const VMConsoles: React.FC<VMConsolesProps> = ({
       <VMIsDown />
     );
   }
-
   const vncServiceManual = (vnc && vnc.manual) || undefined;
```

Comment:
we should check check if vnc / serial is enabled in the VMI and offer only options which are available and potentially override type if the desired option is not available

Reply:
what do you mean by offer? Falling back to vnc will confuse the user, there is no rendering error if the type doesn't exist. the user will see an empty console and will understand that the specific type isn't available.

Reply:
we could remove the vnc/serial options by checking https://kubevirt.io/api-reference/master/definitions.html#_v1_devices

Reply:
remove it from where? these are URL params. The user can enter anything he wants, either it exists or it fallbacks to vnc.
if the type exists and the vm doesn't support it, ill add an alert saying this option is not available.

Reply:
why does it make sense to show VNC/Serial options when the VM doesn't offer them?

We can do the alert when the type is in url. But for the missing type param we can simply switch to the next available option without an alert.